backend/src/main/scala/com/softwaremill/adopttapir/starter/content/DirectoryMerger.scala

```diff
@@ -1,44 +1,44 @@
 package com.softwaremill.adopttapir.starter.content
 
-object DirectoryMerger:
+import cats.syntax.all.*
 
-  def apply(topDirName: String, pathNames: List[String], content: String): Directory =
-    assert(pathNames.nonEmpty, "`pathNames` list cannot be empty!")
+object DirectoryMerger:
 
-    val pathsWithRootDir = List(topDirName) ++ pathNames
-    createTreeRec(pathsWithRootDir, content) match {
-      case File(_, _)          => throw new IllegalStateException("Top element of created tree cannot be a `File`!")
-      case d @ Directory(_, _) => d
+  def apply(topDirName: String, pathNames: List[String], content: String): Either[IllegalStateException, Directory] = for
+    _ <- Either
```

Comment:
Hm I'm not sure about this syntax. Isn't a simple `if` + `Left`/`Right` simpler to read here? We wouldn't have to lift everything to a for-comprehension